examples/morpheusvm/tests/e2e/transfer_test.go

```diff
@@ -0,0 +1,58 @@
+// Copyright (C) 2024, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package e2e_test
+
+import (
+	"context"
+	"time"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/ava-labs/hypersdk/api/jsonrpc"
+	"github.com/ava-labs/hypersdk/auth"
+	"github.com/ava-labs/hypersdk/chain"
+	"github.com/ava-labs/hypersdk/crypto/ed25519"
+	"github.com/ava-labs/hypersdk/examples/morpheusvm/actions"
+	"github.com/ava-labs/hypersdk/examples/morpheusvm/vm"
+	"github.com/ava-labs/hypersdk/tests/workload"
+
+	he2e "github.com/ava-labs/hypersdk/tests/e2e"
+	ginkgo "github.com/onsi/ginkgo/v2"
+)
+
+var _ = he2e.RegisterTest("Transfer Transaction", func(t ginkgo.FullGinkgoTInterface, tn workload.TestNetwork) error {
+	require := require.New(t)
+
+	uri := tn.URIs()[0]
+	other, err := ed25519.GeneratePrivateKey()
+	require.NoError(err)
+	aother := auth.NewED25519Address(other.PublicKey())
+
+	lcli := vm.NewJSONRPCClient(uri)
+	parser, err := lcli.Parser(context.Background())
+	require.NoError(err)
+
+	spendingKey, err := tn.WorkloadFactory().GetSpendingKey()
```

Comment:
in these custom e2e tests, we only use the `workloadfactory` for this one method. This means if the user wants to only create custom e2e tests they need to also define the `workloadfactory` interface which is quite confusing 😅

Reply:
maybe its worth adding some of the further work you mentioned to this PR. If the `GenesisKeys` are added to the test network then we wouldn't need to couple the `WorkloadFactory` here